language/types_values_variables.md

```diff
@@ -456,6 +461,333 @@ For conversion from `String` both float and integer formats are supported.
+a conversion specifier as follows:
+```
+%[Flags][Width]Conversion
+```
+
+###### Flags:
+
+| Flag  | Meaning
+| ----  | ---------------
+| -     | Don't pad numerical output
+| _     | Use spaces for padding
+| 0     | Use zeros for padding
+
+###### Format directives:
+
+| Format | Meaning |
+| ------ | ------- |
+| D | Number of Days |
+| H | Hour of the day, 24-hour clock |
+| M | Minute of the hour (00..59) |
+| S | Second of the minute (00..59) |
+| L | Millisecond of the second (000..999) |
+| N | Fractional seconds digits |
+
+The format directive that represents the highest magnitude in the format will be allowed to
+overflow. I.e. if no "%D" is used but a "%H" is present, then the hours may be more than 23.
+
+The default array contains the following patterns:
+
+```
+['%D-%H:%M:%S', '%D-%H:%M', '%H:%M:%S', '%H:%M']
+```
+
+Examples - Converting to Timespan
+
+```puppet
+$duration = Timespan(13.5)       # 13 seconds and 500 milliseconds
+$duration = Timespan({days=>4})  # 4 days
+$duration = Timespan(4, 0, 0, 2) # 4 days and 2 seconds
+$duration = Timespan('13:20')    # 13 hours and 20 minutes (using default pattern)
+$duration = Timespan('10:03.5', '%M:%S.%L') # 10 minutes, 3 seconds, and 5 milli-seconds
+$duration = Timespan('10:03.5', '%M:%S.%N') # 10 minutes, 3 seconds, and 5 nano-seconds
+```
+
+### Timestamp ([from, to])
+
+Since version 4.8.0
+
+Represents a range of timestamp values. The default is the range +/- Infinity.
+
+A timestamp is an moment in time, measured in seconds since epoch (1970-01-01 00:00:00 UTC), with nano-second precision.
+
+A `Timestamp` range behaves as an `Integer` range and accepts integer, float, or timestamp values when
+specifying the range. It is however, not possible to iterate over a `Timestamp` range.
+
+#### Type Algebra on Timestamp
+
+    Timestamp ∪ Timestamp           → Timestamp
+    Timestamp ∪ Integer             → Numeric
+    Timestamp ∪ Numeric             → Numeric
+    Timestamp ∪ (T ∈ Scalar)        → Scalar
+    Timestamp ∪ (T ∉ Scalar)        → Any
+    Timestamp[a, b] ∪ Timestamp[c, d]   → Timestamp[min(a, c), max(b,d)]
+
+#### Timestamp.new
+
+A new `Timestamp` can be created from `Integer`, `Float`, `String`, and `Hash` values. Several variants of the constructor are provided.
+
+##### Timestamp from seconds since epoch (1970-01-01 00:00:00 UTC)
+
+When a Float is used, the decimal part represents fractions of a second.
+
+```puppet
+function Timestamp.new(
+  Variant[Float, Integer] $value
+)
+```
+
+##### Timestamp from String and patterns consisting of format directives
+
+The first argument is parsed using the format optionally passed as a string or array of strings. When an array is used, an attempt
+will be made to parse the string using the first entry and then with each entry in succession until parsing succeeds. If the second
+argument is omitted, an array of default formats will be used.
+
+An exception is raised when no format was able to parse the given string.
```

Comment:
Use of "exception" (and I checked, we do not use the term in the specification).